### PR TITLE
privacy enhancing patterns

### PR DESCRIPTION
**Pull request by SylTi**
What do you guys think?
Any other patterns in minds? @aantonop 
#85

### CODE REVIEW

File: preface.asciidoc [hunk 1]
```diff
@@ -187,3 +187,4 @@ Following is an alphabetically sorted list of notable GitHub contributors, inclu
 * Yeramin Santana (ysfdev)
 * Zhen Wang (zmxv)
 * ztz (zt2)
+* Sylvain Tissier (SylTi)
```
Comment (zaq1tomo) [inline preface.asciidoc:190]: I think these names sorted by alphabet.

Comment (aantonop) [reply preface.asciidoc:190]: Don't worry, I'll resort them from time to time. It is better to add to the end for now, so that I can track merges easier.

**Comment by aantonop** [comment]
This is an interesting pattern. I'd like some more people to review the contracts themselves before I merge. Comments please?

**Comment by SylTi** [comment]
I have some implementations tests in JS if you're interested.

edit: Also the `PrivacyJudge` is NOT production ready, for example, if user2 never deposit funds inside the judge, user1 funds are locked forever (but this can easily be solved by adding conditions). They are probably other attacks scenarios possibles.

**Pull request merged**